packages/cardinal-integration/src/cardinal-three-d-secure-flow.spec.ts

```diff
@@ -0,0 +1,221 @@
+import { Response } from '@bigcommerce/request-sender';
+import { merge } from 'lodash';
+
+import {
+    HostedForm,
+    OrderRequestBody,
+    PaymentIntegrationService,
+    PaymentMethod,
+    PaymentRequestOptions,
+    PaymentStrategy,
+    RequestError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+import {
+    getBillingAddress,
+    getCheckout,
+    getOrder,
+    getOrderRequestBody,
+    getPaymentMethod,
+    getResponse,
+    getShippingAddress,
+    PaymentIntegrationServiceMock,
+} from '@bigcommerce/checkout-sdk/payment-integrations-test-utils';
+
+import CardinalClient from './cardinal-client';
+import CardinalThreeDSecureFlow from './cardinal-three-d-secure-flow';
+
+describe('CardinalThreeDSecureFlow', () => {
+    let cardinalClient: Pick<
+        CardinalClient,
+        'configure' | 'getThreeDSecureData' | 'load' | 'runBinProcess'
+    >;
+    let threeDSecureFlow: CardinalThreeDSecureFlow;
+    let paymentMethod: PaymentMethod;
+    let paymentIntegrationService: PaymentIntegrationService;
+
+    beforeEach(() => {
+        paymentMethod = {
+            ...getPaymentMethod(),
+            clientToken: 'foo',
+        };
+
+        cardinalClient = {
+            configure: jest.fn(() => Promise.resolve()),
+            getThreeDSecureData: jest.fn(() => Promise.resolve()),
+            load: jest.fn(() => Promise.resolve()),
+            runBinProcess: jest.fn(() => Promise.resolve()),
+        };
+
+        paymentIntegrationService = new PaymentIntegrationServiceMock();
+
+        jest.spyOn(paymentIntegrationService.getState(), 'getPaymentMethodOrThrow').mockReturnValue(
+            paymentMethod,
+        );
+
+        threeDSecureFlow = new CardinalThreeDSecureFlow(
+            paymentIntegrationService,
+            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
Do not forget to remove it when you add this rule to `eslintrc.json`

Reply:
Done